### ISSUE TITLE
False positive with `Style/RedundantEach` cop

### ISSUE THREAD

**Issue by anthony-robin** [rubocop/rubocop#11165]
Given the folliwing Ruby code

```ruby
string.each_char
      .map(&:to_i)
      .reverse
      .each_with_index do |item, index|
  puts item, index
end
```

Rubocop throws an offense

```shell
tmp/test_rubocop.rb:4:8: C: [Correctable] Style/RedundantEach: Use with_index to remove redundant each.
      .each_with_index do |item, index|
       ^^^^^^^^^^^^^^^
```

and autocorrects it to

```ruby
string.each_char
      .map(&:to_i)
      .reverse
      .with_index do |item, index|
  puts item, index
end
```

which is a false positive according to the following error:

```
 NoMethodError:
        undefined method `with_index' for [1, 3, 1, 0, 0, 8, 8, 9, 6, 9, 5, 6, 0, 7]:Array

                  .with_index do |digit, idx|
                  ^^^^^^^^^^^
        Did you mean?  with_options
```

Thank you :)

## RuboCop version

```
$ [bundle exec] rubocop -V
1.38.0 (using Parser 3.1.2.1, rubocop-ast 1.23.0, running on ruby 3.1.2) [x86_64-darwin21]
  - rubocop-performance 1.15.0
  - rubocop-rails 2.17.2
  - rubocop-rspec 2.15.0
```
